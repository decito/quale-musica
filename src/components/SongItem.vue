<script>
export default {
  name: 'SongItem',
  props: {
    song: {
      type: Object,
      required: true,
    },
  },
}
</script>

<template>
  <li
    class="flex justify-between items-center p-3 pl-6 cursor-pointer transition
    duration-300 hover:bg-gray-50"
  >
    <div>
      <a href="#" class="font-bold block text-gray-600">
        {{ song.modifiedName }}
      </a>

      <span class="text-gray-500 text-sm">
        {{ song.displayName }}
      </span>
    </div>

    <div class="text-gray-600 text-lg">
      <span class="comments">
        <i class="fa fa-comments text-gray-600" />
        {{ song.commentCount }}
      </span>
    </div>
  </li>
</template>
