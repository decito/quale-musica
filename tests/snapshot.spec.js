import { RouterLinkStub, shallowMount } from "@vue/test-utils";
import SongItem from "@/components/SongItem.vue";

describe("Snapshot of SongItem.vue", () => {
  it("should render correctly", () => {
    const song = {
      docID: "123foobartaz321",
      modifiedName: "Modified Name",
      displayName: "Display Name",
      commentCount: 13,
    };

    const wrapper = shallowMount(SongItem, {
      propsData: { song },
      global: {
        components: {
          "router-link": RouterLinkStub,
        },
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
