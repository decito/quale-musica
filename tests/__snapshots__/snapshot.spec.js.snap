// Vitest Snapshot v1

exports[`Snapshot of SongItem.vue > should render correctly 1`] = `
<li
  class="flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50"
  id="song-123foobartaz321"
>
  <div>
    <router-link-stub
      class="font-bold block text-gray-600"
      custom="false"
      to="[object Object]"
    />
    <span
      class="text-gray-500 text-sm"
    >
      Display Name
    </span>
  </div>
  <div
    class="text-gray-600 text-lg"
  >
    <router-link-stub
      custom="true"
      to="[object Object]"
    />
  </div>
</li>
`;
